pkg/controller/deliverable/reconciler.go

```diff
@@ -131,26 +136,21 @@ func (r *Reconciler) completeReconciliation(deliverable *v1alpha1.Deliverable, e
 		deliverable.Status.ObservedGeneration = deliverable.Generation
 		updateErr = r.Repo.StatusUpdate(deliverable)
 		if updateErr != nil {
-			r.logger.Error(updateErr, "update error")
-			if err == nil {
-				return ctrl.Result{}, fmt.Errorf("update deliverable status: %w", updateErr)
-			}
+			return ctrl.Result{}, fmt.Errorf("update deliverable status: %w", updateErr)
 		}
 	}
 
-	if err != nil {
+	if err != nil && controller.IsUnhandledError(err) {
 		return ctrl.Result{}, err
 	}
 
+	r.logger.Info("handled error", "error", err)
```

Comment:
should this be inside a branch for `if err != nil`?

e.g.

```go
if err != nil {
  if controller.IsUnhandledError(err) {
    return ctrl.Result{}, err
  }
  r.logger.Info("handled error", "error", err)
}
```

otherwise I imagine we'd be printing `handled error: nil` quite a bit

Reply:
good catch